unittests/test_job_runner.py

```diff
@@ -34,34 +34,42 @@ class TestJobRunner(unittest.TestCase):
     def test_generate_summary(self):
         """ Checks if generate summary functions runs successfully"""
 
-        job_runner_object = JobRunner('https://test.com/', 'v1.0', [])
+        job_runner_object = JobRunner('https://test.com/', '1.0.0', [])
```

Comment:
I think it would really make sense to make use of parametrization in your tests here so that you can run checks against multiple versions. Same goes for all other tests in this and the next module.

Check https://docs.pytest.org/en/6.2.x/parametrize.html